src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,8 +1,97 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Locale;
+
 public class SalaryInfo {
+    private static final int quantityOfSubstrings = 4;
+    private Date dateFrom;
+    private Date dateTo;
+    private SimpleDateFormat format;
+    private ArrayList<EmployeeData> employeeData;
 
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        checkOfValidateData(names, data);
+        String salaryInfo = "";
+        salaryInfo += getSalaryPeriod(dateFrom, dateTo);
+        parsingData(data, names);
+        for (String name : names) {
+            int res = 0;
+            for (EmployeeData employeeDatum : employeeData) {
+                if (name.equals(employeeDatum.name)) {
+                    res += employeeDatum.hours * employeeDatum.income;
+                }
+            }
+            salaryInfo += "\n" + name + " - " + res;
+        }
+        return salaryInfo;
+    }
+
+    public void checkOfValidateData(String [] names, String[] data) {
```

Comment:
```suggestion
    private void validateNamesAndData(String[] names, String[] data) {
```
helper methods should be private
also, pay attention at the method names